phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java

```diff
@@ -507,9 +507,6 @@ public Expression visitLeave(LikeParseNode node, List<Expression> children) thro
             }
             if (index == -1) {
                 String rhsLiteral = LikeExpression.unescapeLike(pattern);
-                if (lhsMaxLength != null && lhsMaxLength != rhsLiteral.length()) {
```

Comment:
Looks good, but just confirm that this does cause any regressions to existing like conditions that have wildcards?

Reply:
Yes. :)